Documentation/deploy-master.md

```diff
@@ -301,6 +286,86 @@ spec:
       timeoutSeconds: 1
 ```
 
+### Run the per host calico agent
+The per host Calico agent runs on all hosts including the master node. It is responsible for enforcing networking policy and connecting the pods on that host to the flannel overlay network.
+
+When creating `/etc/systemd/system/calico-node.service`:
+
+* Replace `${ETCD_ENDPOINTS}`
+* Replace `${ADVERTISE_IP}` with this node's publicly routable IP.
+
+**/etc/systemd/system/calico-node.service**
+
+```yaml
+[Unit]
+Description=Calico per-host agent
+Requires=network-online.target
+After=network-online.target
+
+[Service]
+Slice=machine.slice
+Environment=CALICO_DISABLE_FILE_LOGGING=true
+Environment=HOSTNAME=${ADVERTISE_IP}
+Environment=IP=${ADVERTISE_IP}
+Environment=FELIX_FELIXHOSTNAME=${ADVERTISE_IP}
+Environment=CALICO_NETWORKING=false
+Environment=NO_DEFAULT_POOLS=true
+Environment=ETCD_ENDPOINTS=${ETCD_ENDPOINTS}
+ExecStart=/usr/bin/rkt run --inherit-env --stage1-from-dir=stage1-fly.aci \
+--volume=modules,kind=host,source=/lib/modules,readOnly=false \
+--mount=volume=modules,target=/lib/modules \
+--trust-keys-from-https quay.io/calico/node:v0.19.0
+
+KillMode=mixed
+Restart=always
+TimeoutStartSec=0
+
+[Install]
+WantedBy=multi-user.target
+```
+
+### Set Up the policy-agent Pod
+
+The policy agent is the last major piece of the our master node. It monitors the API for changes related to network policy and configures Calico to implement that policy.
+
+When creating `/etc/kubernetes/manifests/policy-agent.yaml`:
+
+* Replace `${ETCD_ENDPOINTS}`
+
+**/etc/kubernetes/manifests/policy-agent.yaml**
+
+```yaml
+apiVersion: v1
+kind: Pod 
+metadata:
+  name: calico-policy-agent
+  namespace: calico-system 
+spec:
+  hostNetwork: true
```

Comment:
Why does the agent need to be a static pod on the controller with host networking? It looks to me like this pod should be scheduled via a deployment, and we don't need the host networking.

* `ETCD_ENDPOINTS` is routable from the pod network on any kubelet in the cluster. 

* `policy-agent` pod talk to the "in-cluster" kubernetes service at `kubernetes.default.svc.cluster.local` instead of using the insecure port and host networking. This is the standard way that cluster components communicate with the api server.